--- C:\strawberry_build\build\perl_core\perl-5.16.0\win32\makefile.mk.backup	Mon May 14 21:49:22 2012
+++ C:\strawberry_build\build\perl_core\perl-5.16.0\win32\makefile.mk	Wed May 16 14:15:29 2012
@@ -72,6 +72,9 @@
 #
 USE_ITHREADS	*= define
 
+#XXX kmx hack
+USE_64_BIT_INT	*= define
+
 #
 # uncomment to enable the implicit "host" layer for all system calls
 # made by perl.  This needs USE_MULTI above.  
@@ -335,6 +338,10 @@
 
 .IF "$(USE_ITHREADS)" == "define"
 ARCHNAME	!:= $(ARCHNAME)-thread
+.ENDIF
+
+.IF "$(USE_64_BIT_INT)" == "define"
+ARCHNAME	!:= $(ARCHNAME)-64int
 .ENDIF
 
 # Visual C++ 98, .NET 2003, 2005/2008/2010 specific.
