--- C:\strawberry_build\build\perl_core\perl-5.16.0\win32\config_H.gc64nox.backup	Wed Apr 25 02:18:36 2012
+++ C:\strawberry_build\build\perl_core\perl-5.16.0\win32\config_H.gc64nox	Mon May  7 21:28:44 2012
@@ -645,7 +645,7 @@
  *	This symbol, if defined, indicates that <rpcsvc/dbm.h> exists and
  *	should be included.
  */
-/*#define I_DBM	/ **/
+#define I_DBM	/**/
 #define I_RPCSVC_DBM	/**/
 
 /* I_DLFCN:
@@ -670,7 +670,7 @@
  *	This symbol, if defined, indicates that <gdbm.h> exists and should
  *	be included.
  */
-/*#define I_GDBM	/ **/
+#define I_GDBM	/**/
 
 /* I_LIMITS:
  *	This symbol, if defined, indicates to the C program that it should
@@ -2961,7 +2961,7 @@
  *	parameter information. While ANSI C prototypes are supported in C++,
  *	K&R style function declarations will yield errors.
  */
-/*#define I_NDBM	/ **/
+#define I_NDBM	/**/
 /*#define I_GDBMNDBM	/ **/
 /*#define I_GDBM_NDBM	/ **/
 /*#define NDBM_H_USES_PROTOTYPES	/ **/
