--- C:\strawberry_build\build\perl_core\perl-5.22.0-RC1\win32\win32.c.backup	Wed May 13 22:19:31 2015
+++ C:\strawberry_build\build\perl_core\perl-5.22.0-RC1\win32\win32.c	Thu May 14 19:32:22 2015
@@ -1124,6 +1124,7 @@
  * XXX this needs strengthening  (for PerlIO)
  *   -- BKS, 11-11-200
 */
+#if !defined(__MINGW64_VERSION_MAJOR) || __MINGW64_VERSION_MAJOR < 4
 int mkstemp(const char *path)
 {
     dTHX;
@@ -1144,6 +1145,7 @@
 	goto retry;
     return fd;
 }
+#endif
 
 static long
 find_pid(pTHX_ int pid)
