--- C:\strawberry_build\build\perl_core\perl-5.14.2\win32\config_H.gc.backup	Mon Sep 26 11:44:34 2011
+++ C:\strawberry_build\build\perl_core\perl-5.14.2\win32\config_H.gc	Sun Apr 22 20:41:55 2012
@@ -649,7 +649,7 @@
  *	This symbol, if defined, indicates that <rpcsvc/dbm.h> exists and
  *	should be included.
  */
-/*#define I_DBM	/ **/
+#define I_DBM	/**/
 #define I_RPCSVC_DBM	/**/
 
 /* I_DLFCN:
@@ -674,7 +674,7 @@
  *	This symbol, if defined, indicates that <gdbm.h> exists and should
  *	be included.
  */
-/*#define I_GDBM	/ **/
+#define I_GDBM	/**/
 
 /* I_LIMITS:
  *	This symbol, if defined, indicates to the C program that it should
@@ -2435,7 +2435,7 @@
  *	parameter information. While ANSI C prototypes are supported in C++,
  *	K&R style function declarations will yield errors.
  */
-/*#define I_NDBM	/ **/
+#define I_NDBM	/**/
 /*#define I_GDBMNDBM	/ **/
 /*#define I_GDBM_NDBM	/ **/
 /*#define NDBM_H_USES_PROTOTYPES	/ **/
