--- C:\strawberry_build\build\perl_core\perl-5.14.2\win32\FindExt.pm.backup	Mon Sep 19 15:18:22 2011
+++ C:\strawberry_build\build\perl_core\perl-5.14.2\win32\FindExt.pm	Thu Nov  3 10:37:18 2011
@@ -5,8 +5,7 @@
 use strict;
 use warnings;
 
-my $no = join('|',qw(GDBM_File ODBM_File NDBM_File DB_File
-                     VMS VMS-DCLsym VMS-Stdio Sys-Syslog IPC-SysV I18N-Langinfo));
+my $no = join('|',qw(DB_File VMS VMS-DCLsym VMS-Stdio Sys-Syslog IPC-SysV I18N-Langinfo));
 $no = qr/^(?:$no)$/i;
 
 sub apply_config {
